packages/generic/wirisplugin-generic.src.js

```diff
@@ -119,22 +121,25 @@ export default class GenericIntegration extends IntegrationModel {
      *
      * @param {string} toolbar The MT/CT button clicked from the toolbar: 'general' for the MathType and 'chemistry' for the ChemType
      * @param {string} trigger 'mtct_insert' when the modal is closed due to inserting or modifying a formula. 'mtct_close' otherwise
+     ** The function was changed to async to ensure that telemetry tracking is completed before closing the modal.
      */
-    wrsClosedEditorModal(toolbar, trigger) {
+    async wrsClosedEditorModal(toolbar, trigger) {
       // Check that the manual string inputs contain the values we want, if not throw error.
-      if (/^(mtct_insert|mtct_close)$/.test(trigger) && /^(general|chemistry)$/.test(toolbar)) {
-        // Call Telemetry service
-        try {
-          Telemeter.telemeter.track("CLOSED_MTCT_EDITOR", {
-            toolbar: toolbar,
-            trigger: trigger,
-          });
-        } catch (err) {
-          console.error(err);
-        }
-      } else {
+      if (!/^(mtct_insert|mtct_close)$/.test(trigger) || !/^(general|chemistry)$/.test(toolbar)) {
         console.error('Invalid trigger or toolbar value for close editor modal');
+        return;
+      }
+
+      try {
+        await Telemeter.telemeter.track("CLOSED_MTCT_EDITOR", {
+          toolbar: this.contentManager.toolbar,
+          trigger: "mtct_close",
+        });
+      } catch (error) {
+        console.error("Error tracking CLOSED_MTCT_EDITOR", error);
       }
+
+      this.close();
```

Comment:
I can't see any `close` method neither in this class nor in `IntegrationModel`, which this class extends. I'm pretty sure this should fail, am I missing something?